quantum/pointing_device.c

```diff
@@ -71,13 +76,66 @@ __attribute__((weak)) void pointing_device_send(void) {
 }
 
 __attribute__((weak)) void pointing_device_task(void) {
+#if defined(SPLIT_POINTING_ENABLE)
+    // Don't poll the target side pointing device.
+    if (!is_keyboard_master()) {
+        return;
+    };
+#endif
+
+#if defined(POINTING_DEVICE_TASK_THROTTLE)
+    static uint32_t last_exec = 0;
+    if (timer_elapsed32(last_exec) < 1) {
+        return;
+    }
+    last_exec = timer_read32();
+#else
+#    if defined(SPLIT_POINTING_ENABLE)
+#        pragma message("It's recommended you enable a throttle when sharing pointing devices.")
+#    endif
+#endif
+
     // Gather report info
 #ifdef POINTING_DEVICE_MOTION_PIN
+#    if defined(SPLIT_POINTING_ENABLE)
+#        error POINTING_DEVICE_MOTION_PIN not supported when sharing the pointing device report between sides.
+#    endif
     if (!readPin(POINTING_DEVICE_MOTION_PIN))
 #endif
-        mouseReport = pointing_device_driver.get_report(mouseReport);
 
-        // Support rotation of the sensor data
+#if defined(SPLIT_POINTING_ENABLE)
+#    if defined(POINTING_DEVICE_COMBINED)
+        static report_mouse_t sharedReport = {0};
+    static uint8_t oldButtons = 0;
+    mouseReport.buttons       = oldButtons;
+    mouseReport               = pointing_device_driver.get_report(mouseReport);
+    sharedReport              = get_targets_pointing();
+    oldButtons                = mouseReport.buttons;
+    mouseReport.x             = mouseReport.x | sharedReport.x;
+    mouseReport.y             = mouseReport.y | sharedReport.y;
+    mouseReport.h             = mouseReport.h | sharedReport.h;
+    mouseReport.v             = mouseReport.v | sharedReport.v;
+    mouseReport.buttons       = mouseReport.buttons | sharedReport.buttons;
```

Comment:
May be best to add a handler here, that is weakly defined.  That way, functionality can be customized (eg, one side for mouse movement, and one side for scrolling). 

In fact, it may be worth moving the whole split pointing enable part to a separate function?

Reply:
I was thinking of implement a separate function and probably introducing *_RIGHT defines. So the pointing devices processed as LEFT/RIGHT vs MASTER/SLAVE so it's consistent.